File: demo/ex_3.cpp
/*  Example 3: Memory Management

    Compile ->  g++ ex_3.cpp
    Run ->      ./a.out
*/

#include <iostream>
#include <string>

int main()
{
    {
        std::string* string_1 = new std::string("Stay Froggy");  // In scope
    }

    // Out of scope but not deleted!
}

File: demo/ex_3.rs
/*  Example 3: Memory Management

    Compile ->  rustc ex_3.rs
    Run ->      ./ex_3
*/

fn main() {
    
    {
        let string_1 = String::from("Stay Froggy!");  // In scope
    }

    // Out of scope and AUTOMAGICALLY deleted

}